Add Hookdeck CLI quickstart

leggetter opened the pull request:
~I do still need to give this a once over, but~ I'd love feedback on this, @ajot.

I'm not sure about the name of the quickstart. It's a little long.

The quickstart itself is also potentially a little long.

I've used images as part of using the CLI and the Hookdeck Console is quite visual.

---

The [Hookdeck CLI](https://github.com/hookdeck/hookdeck-cli) is a free -- no account required -- alternative to ngrok for asynchronous web development: perfect for receiving inbound webhooks on your localhost.

leggetter committed:
first draft of Hookdeck CLI quickstart

leggetter committed:
Merge branch 'main' of https://github.com/ajot/hello-world-guides-mintlify into hookdeck-cli

leggetter committed:
hookdeck typos/grammar

leggetter committed:
hookdeck read-through

ajot reviewed (commented):
Hey Phil,

Thanks for submitting this PR. I followed the guide and, after a bit of a struggle, got my webhook up and running. Here are some suggestions:

1. **Include a Simple curl Command**: Once I had Hookdeck CLI setup, getting the tunnel going was quick. I wanted to fire off a simple curl command to trigger the webhook. I think we should include this step upfront. How about we move the example webhooks to an “optional” section? They’re great, but not as clear as a straightforward "hello world" curl statement which is really the first step.

2. **Clarify Flask Route Setup**: The Flask route threw me off while I was trying to understand why my curl statements weren’t working. All my requests were being routed to `/webhook`. I then recalled that this was set up when creating the tunnel. I  didn't understand why at that point. We should perhaps drill this point home during the setup process that all requests will be directed to `/webhook`. 

3. **Simplify Route Definition**: The catch-all route (`@app.route("/<path:path>", methods=["POST"])`) was confusing. Simplifying this to a specific `/webhook` route makes it much easier to understand. How about we simplify the route to handle POST requests sent to `/webhook` instead of the catch-all version?

4. **Naming Conventions**: How about we name the webhook `my-webhook` to make it clear that this can be anything you want. For the connection label, maybe use `My API` instead of `localhost`.

`hookdeck listen 3030 my-webhook`

5. **Example curl Command and Flask App**: Here’s the curl statement we could include:
    ```sh
    curl -X POST https://hkdk.events/cnzpo680rdhejk -H "Content-Type: application/json" -d '{"message": "Hello, World!", "status": "success"}'
    ```

    And the relevant Flask app:
    ```python
    from datetime import datetime
    import json
    import logging
    from flask import Flask, request

    # Configure logging to output to the console
    logging.basicConfig(
        level=logging.INFO,
        handlers=[logging.StreamHandler()]
    )

    app = Flask(__name__)
    app.logger.setLevel(logging.INFO)

    # Define a route to handle POST requests sent to /webhook
    @app.route("/webhook", methods=["POST"])
    def handle_hello_webhook():
        # Log the received webhook with the current timestamp and JSON payload
        app.logger.info("webhook_received %s %s",
                        datetime.now().isoformat(),
                        json.dumps(request.json, indent=2))

        # Respond with a status of ACCEPTED and message "hello world"
        return {
            "status": "ACCEPTED",
            "message": "hello world"
        }

    if __name__ == "__main__":
        # Run the Flask app on port 3030 in debug mode
        app.run(debug=True, port=3030)
    ```

Thanks again for your hard work on this! Let me know if you have any questions.

PS: I do have some edits from a mintlify format pov (using accordion, and reusable snippets etc, but I can make those updates once you're done).

leggetter committed:
address most feedback

leggetter commented:
Thanks for the detailed feedback, @ajot 🙌 

I've addressed the following in https://github.com/ajot/hello-world-guides-mintlify/pull/7/commits/e2a67608fb804c725b85ac9a33a0464603e6351e:

1. cURL command. I removed the `status` property.
2. Flask route clarification in two places
3. Using `/webhook` as the only route in the Flask app
4. Updated naming
5. Which provides the updates for 1 and 3.

I haven't yet moved the additional sections to an optional section. Do you have an example of how you prefer to do that?

ajot commented:
> Thanks for the detailed feedback, @ajot 🙌
> 
> I've addressed the following in [e2a6760](https://github.com/ajot/hello-world-guides-mintlify/commit/e2a67608fb804c725b85ac9a33a0464603e6351e):
> 
> 1. cURL command. I removed the `status` property.
> 2. Flask route clarification in two places
> 3. Using `/webhook` as the only route in the Flask app
> 4. Updated naming
> 5. Which provides the updates for 1 and 3.
> 
> I haven't yet moved the additional sections to an optional section. Do you have an example of how you prefer to do that?

I think this looks great. I will go ahead, and accept the PR. For the optional stuff, let me take a stab at it, and open a PR for your review. Thank you again for contributing this one 🙌
